### PR TITLE
Align devices update 

### LINKED ISSUE #103: Some not specified fields should not be updated to null
Currently when a PUT request is done and one optional field is not specified it is updated to null. 

A solution should be to add an option boolean to crud.put or crud.update_entry, if set to True it will update only the not-None fields

### PR DESCRIPTION
**Pull request by jeanpasquier75**
Relates to discussion in #45 (devices route) and closes #103 for the route `PUT /devices/{device_id}/`

* Add admin scope (let's say that only admin can change the `owner_id`)
* Use the option `only_provided` defined in `crud.update_entry` that updates only the fields provided in the payload (eg. the admin only updates the `owner_id`, the device still gets its position, etc.)
* Change low level `payload` argument of crud.base.put to dictionary instead of `pydantic.BaseModel` & migrate routes that used the function
* Add the only_provided option that remove None values from payload, retrieve from table record afterwards to get all fields ? Waiting for feedbacks on this one

This is first iteration, more concret than issue comments haha

### CODE REVIEW

**Comment by frgfm** [comment]
Thanks for all this work! 
Two things apart from the review:
- the owner of a device should be able to transfer ownership to someone else. Do you see a way for us to integrate this in the PR?
- when I first read "provided_only" I admit I had no clue what you were about to say. I would suggest replacing with "only_specified"

**Review by frgfm — COMMENTED** [review]
Apart from my previous comments, the rest of the PR looks good to me!

File: src/app/api/crud/accesses.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ async def create_accessed_entry(
     return entry
 
 
-async def update_accessed_entry(table: Table, accesses: Table, entry_id: int, payload: Any):
+async def update_accessed_entry(table: Table, accesses: Table, entry_id: int, payload: Any, only_provided=False):
```
Comment (florianriche) [inline src/app/api/crud/accesses.py:75]: Maybe add a UTest?
IMHO the default boolean should be set to True :)

**Review by florianriche — APPROVED** [review]

File: src/app/api/crud/accesses.py (hunk 1 above)
Comment (jeanpasquier75) [reply src/app/api/crud/accesses.py]: I changed the argument to `only_specified` and set its default value to True

**Review by frgfm — APPROVED** [review]
Thanks a lot!

**Pull request merged**